mobile/lib/shared/services/immich_logger.service.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/foundation.dart';
+import 'package:hive/hive.dart';
+import 'package:immich_mobile/constants/hive_box.dart';
+import 'package:immich_mobile/shared/models/in_app_logger_message.model.dart';
+
+class ImmichLogger {
+  String logContext = "";
+  final Box<InAppLoggerMessage> _box = Hive.box(inAppLoggerBox);
+
+  List<InAppLoggerMessage> get messages =>
+      _box.values.toList().reversed.toList();
+
+  ImmichLogger(this.logContext) {
+    if (_box.length > 100) {
+      var indexesToBeDeleted = _box.length - 100;
+      for (var i = 0; i < indexesToBeDeleted; i++) {
+        _box.deleteAt(indexesToBeDeleted);
```

Comment:
As far as I understand, new elements are appended to the end of the box. So could this be simplified to `_box.deleteAt(0);`?

Reply:
So the constructor only runs when it is instantiated, correct?. If you have more than 150 items on the list, this current implementation will remove the first 50 items

Reply:
`deleteAt` is an indexed delete, so this current code is actually deleting entries from somewhere halfway (at index `indexesToBeDeleted`). To remove the first 50 items, you should call `_box.deleteAt(0);` 50 times instead.

Reply:
Ah duh 😶‍🌫️